### PR TITLE
Update newRobotStateCallback to use the new the DisplayRobotState msg

### PR DESCRIPTION
**Pull request by JafarAbdi**
### Description

This PR is related to [PR#56](https://github.com/ros-planning/moveit_visual_tools/pull/56)

So after updating the DisplayRobotState msg to include a new field [see](https://github.com/ros-planning/moveit_msgs/pull/55), and updating [MoveItVisualTools::hideRobot()](https://github.com/ros-planning/moveit_visual_tools/blob/master/src/moveit_visual_tools.cpp#L1528) to toggle the .hide rather than using a virtual_joint that moves the robot very far away location in Rviz; there's a need to add the feature in the Rviz Display,

I think that one solution could be by updating the robot visibility `robot_->setVisualVisible(!state_msg->hide)` inside [RobotStateDisplay::newRobotStateCallback](https://github.com/ros-planning/moveit/blob/2505cf4e4783386fc828553d60cd6f78accda4bb/moveit_ros/visualization/robot_state_rviz_plugin/src/robot_state_display.cpp#L297)

Another option is to change `enable_visual_visible_->setBool(!state_msg->hide)` directly but this option will change the checkmark, too.
RVIZ_Soln#1

The current solution will not affect it
RVIZ_Soln#2

I tested it with these launch files

`roslaunch moveit_visual_tools demo_rviz.launch`
`roslaunch moveit_visual_tools demo.launch`

### Checklist
- [x] **Required by CI**: Code is auto formatted using [clang-format](http://moveit.ros.org/documentation/contributing/code)
- [ ] Extend the tutorials / documentation [reference](http://moveit.ros.org/documentation/contributing/)
- [ ] Document API changes relevant to the user in the MIGRATION.md notes
- [ ] Create tests, which fail without this PR [reference](https://ros-planning.github.io/moveit_tutorials/)
- [ ] Include a screenshot if changing a GUI
- [ ] While waiting for someone to review your request, please help review [another open pull request](https://github.com/ros-planning/moveit/pulls) to support the maintainers

[//]: # "You can expect a response from a maintainer within 7 days. If you haven't heard anything by then, feel free to ping the thread. Thank you!"

### CODE REVIEW

**Comment by JafarAbdi** [comment]
Apologizes for the unnecessary commits, I don't know why they're included in the PR :)

**Comment by JafarAbdi** [comment]
Thank you for reviewing it, yes it should be robot_->setVisible; seem like using setVisualVisible after it is redundant [see](http://docs.ros.org/kinetic/api/rviz/html/c++/classrviz_1_1Robot.html#a9658b4628beca0c88d04272a94055aea)

**Comment by v4hn** [comment]
Following up on your patch, I fixed two other closely related issues in the plugin.
Please have a look at my additional commits.
- sending a `hide= true` message will now succeed without printing a `ROS_WARN`
- when the display is enabled by the user, it always visualized an empty default state. With the new functionality, we can just remove the additional `setVisible` to get rid of this.

File: moveit_ros/visualization/robot_state_rviz_plugin/src/robot_state_display.cpp [hunk 1]
```diff
@@ -303,7 +304,8 @@ void RobotStateDisplay::newRobotStateCallback(const moveit_msgs::DisplayRobotSta
   // possibly use TF to construct a robot_state::Transforms object to pass in to the conversion function?
   try
   {
-    robot_state::robotStateMsgToRobotState(state_msg->state, *robot_state_);
+    if(!planning_scene::PlanningScene::isEmpty(state_msg->state))
```
Comment (davetcoleman) [inline moveit_ros/visualization/robot_state_rviz_plugin/src/robot_state_display.cpp]: Is this simple utility worth the dependency?

**Comment by davetcoleman** [comment]
Clang format failed, see our code style page on website for quick command to fix

**Comment by v4hn** [comment]
> Clang format failed, see our code style page on website for quick command to fix

Ouch, sorry. I just discovered my pre-commit git hook for fixing style was broken.
I pushed a fixup.

> Is this simple utility worth the dependency?

It's the official location of these checks at the moment and changing that seems over the top for this request.
Notice that this does not add ROS dependencies, as the planning_scene is part of `moveit_core`.
I'll look into moving these checks to `moveit_core/utils`, but not as part of this request.

File: moveit_ros/visualization/robot_state_rviz_plugin/src/robot_state_display.cpp [hunk 2]
```diff
@@ -303,7 +304,8 @@ void RobotStateDisplay::newRobotStateCallback(const moveit_msgs::DisplayRobotSta
   // possibly use TF to construct a robot_state::Transforms object to pass in to the conversion function?
   try
   {
-    robot_state::robotStateMsgToRobotState(state_msg->state, *robot_state_);
+    if (!planning_scene::PlanningScene::isEmpty(state_msg->state))
+      robot_state::robotStateMsgToRobotState(state_msg->state, *robot_state_);
```
Comment (rhaschke) [inline moveit_ros/visualization/robot_state_rviz_plugin/src/robot_state_display.cpp]: That's a great addition as this allows sending just some highlight links.

**Comment by rhaschke** [comment]
In addition to @v4hn's commit mentioned in https://github.com/ros-planning/moveit/pull/1625#issuecomment-521785070, I added a status display to provide a warning message when the display is hidden (or no message was received yet).

**Pull request merged**